guides/01_getting-started/03_sharing-your-app.md

```diff
@@ -166,6 +166,42 @@ demo.launch(auth=same_auth)
 For authentication to work properly, third party cookies must be enabled in your browser.
 This is not the case by default for Safari, Chrome Incognito Mode.
 
+## Enable OAuth
+
+Gradio supports OAuth login via Hugging Face. This feature is currently **experimental** and only available on Spaces.
+If allows to add a *"Sign in with Hugging Face"* button to your demo. Check out [this Space](https://huggingface.co/spaces/Wauplin/gradio-oauth-demo)
+for a live demo.
+
+To enable OAuth, you must set `hf_oauth: true` as a Space metadata in your README.md file. This will register your Space
+as an OAuth application on Hugging Face. Next, you can use `gr.LoginButton` and `gr.LogoutButton` to add login and
+logout buttons to your Gradio app. Once a user is logged in with their HF account, you can retrieve their profile. To
```

Comment:
They also need to include `itsdangerous` and `authlib` in a separate requirements.txt for now, right?

Reply:
Btw @Wauplin would you like to open PRs against the dockerfiles to add? https://github.com/huggingface/spaces-app-manager/tree/main/init_build/dockerfiles

Reply:
> They also need to include itsdangerous and authlib in a separate requirements.txt -- at least until we can include it in the Spaces dockerfile -- right?

Right! I added a line in the docs.

> Btw @Wauplin would you like to open PRs against the dockerfiles to add?

Yes I'm planning to do that but waiting for this PR to be merged and a version of gradio to be released first :+1: I'll take care of the PRs.

Reply:
Opened a PR: https://github.com/huggingface/spaces-app-manager/pull/1037